src/partition.ml

```diff
@@ -0,0 +1,94 @@
+open Encoding
+module H_contents = Irmin.Hash.Typed (Hash) (Contents)
+
+type entry = { name : string; kind : [ `Node | `Contents ]; hash : Hash.t }
+type enc_entry = { eencoding : string; entry : entry }
+type pointer = { index : int; hash : Hash.t }
+type enc_pointer = { pencoding : string; pointer : pointer }
+
+type vs =
+  | Empty
+  | Values of enc_entry list
+  | Tree of {
+      depth : int;
+      length : int;
+      pointers : (enc_pointer * enc_vs) list;
+    }
+
+and enc_vs = { vsencoding : string; vs : vs }
+
+let pre_hash t v =
+  let buf = Buffer.create 13 in
+  let pre_hash = Irmin.Type.(unstage (pre_hash t)) in
```

Comment:
It's better to not use `Irmin.Type.` at all here (if possible).

Reply:
I'm using it to encode integers in LEB128 only, I can even remove the `t` parameter